R/r.R

```diff
@@ -150,6 +150,9 @@ check_file_name <- function(name) {
   if (!is_string(name)) {
     ui_stop("Name must be a single string")
   }
+  if (path_file(name) %in% c(".R", "test-.R", ".cpp", ".c")) {
```

Comment:
This isn't scaleable but I figured our scope is limited and this is easier than regex, particularly since we're most likely adding an extension to `""`

Reply:
Have you been bitten by this in practice? It feels like the check is awkwardly distant / indirect / at a different level from actual user input via argument.

I'm tempted to just leave this edge case unhandled. If it seems really important, can we rephrase it as an immediate and simple check on user input?

Reply:
Believe it or not, I have several times due to a slip of the finger, and it's actually pretty annoying to then clean up. I'll refactor the checks to the front